File: pset1/cash/cash.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

void number_of_coins(int change);

int main(void)
{

    float change;

    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0.00);

    printf("changed owed: %.2f\n", change);

    int cents = round(change * 100);
    printf("cents: %i\n", cents);
    number_of_coins(cents);

}

void number_of_coins(int change)
{
    int no25 = change / 25;
    // printf("no25: %i\n",no25);
    change = change - no25 * 25;
    int no10 = change / 10;
    // printf("no10: %i\n",no10);
    change = change - no10 * 10;
    int no5 = change / 5;
    // printf("no5: %i\n",no5);
    change = change - no5 * 5;
    int no1 = change;
    // printf("no1: %i\n",no1);

    printf("Greedy Algorithm coins: %i\n", no25 + no10 + no5 + no1);
}

File: pset6/cash/cash.py
from cs50 import get_float



while(True):
    change = get_float("Change owed: ")
    if(change > 0.0):
        break

change = change * 100

coins = 0

No25 = (int) (change/25)
change -= No25*25
No10 = (int) (change/10)
change -= No10*10
No5 = (int) (change/5)
change -= No5*5
No1 = (int) (change/1)
change -= No1*1

coins = No25 + No10 + No5 + No1
print(f"coins: {coins}, change: {change}")

